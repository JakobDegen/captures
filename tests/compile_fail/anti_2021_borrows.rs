use captures::*;

// check that `all` directives really do capture *everything*
fn main() {
    let mut a = (0, 0);
    let f = capture!(all a, || a.0 + 1);
    a.1 += 1;
    f();
}
