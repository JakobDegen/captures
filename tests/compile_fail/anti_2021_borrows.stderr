error[E0506]: cannot assign to `a.1` because it is borrowed
 --> tests/compile_fail/anti_2021_borrows.rs:7:5
  |
6 |     let f = capture!(all a, || a.0 + 1);
  |                          -  -- borrow of `a.1` occurs here
  |                          |
  |                          borrow occurs due to use in closure
7 |     a.1 += 1;
  |     ^^^^^^^^ assignment to borrowed `a.1` occurs here
8 |     f();
  |     - borrow later used here
